src/nnet3/nnet-nnet.cc

```diff
@@ -784,5 +784,30 @@ std::string Nnet::Info() const {
   return os.str();
 }
 
+void Nnet::ResetGenerators() {
+  // resets random-number generators for all random
+  // components.
+  for (int32 c = 0; c < NumComponents(); c++) {
+    RandomComponent *rc = dynamic_cast<RandomComponent*>(GetComponent(c));
+    if (rc != NULL)
+      rc->ResetGenerator();
+  }
+}
+
+void Nnet::RemoveDropout() {
```

Comment:
this function would not work.. nnet3 isn't just a linear sequence of components.  Just remove this function for now.